main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+	"go_service/src"
+	"go_service/src/adapters"
+	"go_service/src/api"
+	"go_service/src/common"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"log"
+	"net/http"
+	"os"
+)
+
+func setupHandlers(cr common.CurrencyReader, sa api.SubscriberGateway) *http.ServeMux {
+	sm := http.NewServeMux()
+	sm.HandleFunc("/", api.GetCurrencyHandler(cr))
+	sm.HandleFunc("/subscribe", api.GetSubscribersHandler(sa))
+	return sm
+}
+
+func fetchEnv(name string) string {
+	value := os.Getenv(name)
+	if value == "" {
+		log.Fatal(fmt.Sprintf("Environment variable %s is not set", name))
+	}
+	return value
+}
+
+func main() {
+	currencyRateUrl := fetchEnv("CURRENCY_RATE_URL")
+	currencyCode := fetchEnv("CURRENCY_CODE")
+	dbUser := fetchEnv("POSTGRES_USER")
+	dbPassword := fetchEnv("POSTGRES_PASSWORD")
+	dbName := fetchEnv("POSTGRES_DB")
+	dbPort := fetchEnv("DB_PORT")
+	dbHost := fetchEnv("DB_HOST")
+	email := fetchEnv("EMAIL")
+	emailPassword := fetchEnv("EMAIL_PASSWORD")
```

Comment:
I think it would be better to use these env variables on demand: where you really need them, without declaring them at the beginning of the main function. Please see the following comments about moving code to separate functions. This is where you can use this “on-demand” principle